src/Pos/Generator/Block/Payload.hs

```diff
@@ -228,12 +226,18 @@ genTxPayload = do
                 (txIns, inputsResolved, inputsSum) <- generateInputs randomAttempts expectedFee
                 txOutAuxs <- generateOutputs inputsSum expectedFee
 
-                let resolveSk :: Address -> SecretKey
-                    resolveSk = \case
-                        PubKeyAddress stId _ -> resolveSecret stId
-                        other -> error $
-                            sformat ("Found non-pubkey address: "%addressDetailedF) other
-                let resolvedSks = map (resolveSk . txOutAddress) inputsResolved
+                invAddrSpendingData <- unInvAddrSpendingData <$>
+                    view (blockGenParams . asSpendingData)
+                let addrToSk :: MonadThrow n => Address -> n SecretKey
+                    addrToSk addr = do
+                        spendingData <- maybeThrow (BGUnknownAddress addr)
+                            (invAddrSpendingData ^. at addr)
+                        case spendingData of
+                            PubKeyASD pk -> resolveSecret (addressHash pk)
+                            another -> error $
```

Comment:
Actually no, I don't want to get rid of it :)
It's not mine, so I will leave it :)
If it was mine, I would get rid of it.